rele/contrib/logging_middleware.py

```diff
@@ -70,7 +71,7 @@ def post_publish_failure(self, topic, exception, message):
                     "name": "publications",
                     "data": {"agent": self._app_name, "topic": topic},
                 },
-                "subscription_message": message,
+                "subscription_message": json.dumps(message),
```

Comment:
Why we are not being consistent when casting to string?

`json.dumps(message)` vs `str(message)` in line 116

Reply:
I suggest using `message.data.decode("utf-8")`

Reply:
The difference between json.dumps here is because "message" is in fact the raw "data" developer send when invoke publish(). In this case we don't know if that raw data can be cast to str so we have choose the json representation as we do before send the payload to the google's backend

However in the case of str this is because "message" is a Message object from the google library. In this case the Message object can be safely cast to str because it contains a __repr__ method.

Hope it helps to clarify the reasoning behind of this design decission

Reply:
About the `decode("utf-8")` just mention this is not necessary as the value may not be str and the library does not encode the value before call this webhook.

The value is encoded but before send to the google's backend so it has to be decoded before consume the payload